http-ballerina-tests/tests/http_client_data_binding.bal

```diff
@@ -31,11 +31,19 @@ type ClientDBPerson record {|
     int age;
 |};
 
+type ClientDBPersonOtp ClientDBPerson?;
+
+type ClientDBPersonArrOtp ClientDBPerson[]?;
+
+type StringOpt string?;
+type XmlOpt xml?;
+type ByteOpt byte[]?;
```

Comment:
We need same behaviour for the `json?` also. Shall we add it along with a test case?